### ISSUE TITLE
Don't start minio in the test subprocess

### ISSUE THREAD

**Issue by petrbel** [iterait/shepherd#31]
This is actually really ugly.

Minio is a well separated project (from shepherd) and I can't see any reason to force a developer (me) to have minio installed on my bare metal. On the contrary, it is completely fine to have minio executed in a container on a specified port.

So I suggest:
1. removing executing minio in the test subprocess
2. letting user to handle minio execution (docker example: `docker run -it  -p 9000:9000 minio/minio server --address=0.0.0.0:9000 /export`)

Would love to see some comments on this @Teyras @blazekadam @FloopCZ @gdynusa

**Comment by FloopCZ** [comment]
I agree that tests are run by users that are capable of launching `minio` on their own if they are instructed how to do it via docker or via bare metal executable. So +1 for me.

**Comment by Teyras** [comment]
What I don't like about requiring devs to launch minio is that you have to tell them to do it somehow. This is something a developer trying to start hacking on the project will surely stumble on (it happens to the best of us).

To the point of requiring to have minio installed... Well, running it inside a container is pretty much the same thing anyway (at least on Arch). What I like about the tests starting it automatically is that they can make sure there is no unexpected stuff in the storage without breaking anything (because they run their own instance in their own directory).

A traditional approach to this problem would be to mock the Minio client. This would probably take quite some time to do properly. An apparent benefit of this is that you wouldn't have to install the Minio server at all. However, Minio is still required to do *anything* with shepherd (until we decide to implement some other storage backends). A less obvious advantage is that you could easily test scenarios where the Minio server breaks down unexpectedly (this is currently handled rather poorly).

TL;DR - until someone writes a mock implementation of the Minio client, we should stick to launching the server in a subprocess

**Comment by blazekadam** [comment]
I agree with @Teyras completely.

The test process should handle everything for you and requiring certain package/third-party SW for the development is valid.
